Static Assets - added section for public folder (#261)
withastro/docs: src/pages/en/guides/imports.md

- Markdown (`.md`)
- WASM (`.wasm`)

## `public/`
Any files in your `public/` directory are copied into the final build, untouched by Vite or Astro. 

Reference them in your Astro component template via absolute URL paths, from the `public/` directory.

```
├── public/
│   ├── images/
|   |   └── photo1.jpg
│   ├── styles/
|   |   └── global.css
|   └── utils.js
```

```astro
---

> // Astro component script

Reviewer: If there's nothing in the front matter, it's safe to remove:

```diff
- ---
- // Astro component script
- ---
<head>
  <link rel="stylesheet" href="/styles/global.css">
</head>
<body>
  <img src="/images/photo1.jpg">
  <script type="module" src="/nav-utils.js"></script>
</body>
```

---
<head><link rel="stylesheet" href="/styles/global.css"></head>
<body>
  <img src="/images/photo1.jpg"> 

  <script type="module" src="/nav-utils.js"></script>
</body>

```

> ## `src/`

Reviewer: I don't think that this change should force all the existing headers to be nested. I still think the main organization of this page is grouped "by file type".

Can we keep the rest of this page as-is, and instead rename the `public/` header to something like `public/ vs. src/` or `Static Assets in public/`. 

And then rephrase the section/intro a bit to highlight that the rest of the page is about ESM imports, which only work in the `src/` directory.

Author: Made a tweak to see what you think: I've put public at the bottom now, emphasizing that the page is mostly about src files, but with a quick link to the bottom.

Reviewer: I took a pass and rewrote a bit more. Kept the `public/` reference and moved it higher onto the page, but also made it shorter / less highlighted. I think it works best as a "hey FYI public exists" mention but not a full-fledged "Here's how public works" given that we've already covered that in other parts of the site.

Author: No problem! Sorry I kept getting the intent of this page wrong. I was motivated by having a clear `publc/` vs `/src` comparison, as we do see this come up in support, so I thought this might be a page to do that. 

But, I think this can make for its own content (eventually) somewhere else. My `public/` blurb will just go off and join the others in the Land of Misfit Examples for future, more tutorial-like material.  :)

Fixed a pre-existing typo, and otherwise, page looks great!

Reviewer: lets see if this improves the support issues that you see, happy to take another look if it becomes an issue.

My hope is that this makes that point clear, but another thing that we could do is to add some bold to the paragraph, it make it pop more:

> Remember that you can also place any static asset in the [`public/` directory](/en/core-concepts/project-structure/#public) of your project, and Astro will copy the directly into your final build. `public/` files are not built or bundled by Astro, which means that any type of file is supported. You can reference a `public/` file by a URL path directly in your HTML templates.

>  The following applies to files in your `src/` directory, which Astro is ultimately responsible for.

Reviewer 2: ```suggestion
 The following applies to files that are located within your `src/` directory, which Astro is wholly responsible for.
```

Author: Actually, I like *neither* of those "Astro is responsible for" options.  :)

What about changing that to something descriptive:

"The following applies to files located within your `src/` directory, which are processed by Astro."  ??

>  The following applies to files in your `src/` directory, which Astro is ultimately responsible for.

Author: ```suggestion
 The following applies to files located in your `src/` directory, which are processed by Astro.
```

### JavaScript & ESM

> Astro was designed for JavaScript’s native ES Module (ESM) syntax. ESM lets you define explicit imports & exports that browsers and build tools can better understand and optimize for. If you're familiar with the `import` and `export` keywords in JavaScript, then you already know ESM!

Reviewer 2: ```suggestion
Astro was designed for JavaScript’s native ES Module (ESM) syntax. ESM lets you define explicit _imports_ & _exports_ that browsers and build tools can better understand and optimize for. If you're familiar with the `import` and `export` keywords in JavaScript, then you already know ESM!
```

Author: I don't think this helps the paragraph since import and export are inlined later. Italics isn't the *best* for accessibility (bold is better) and I'm not sure I'd do either one here, since I think "regular words" suit this description, followed up by the inline emphasis.

> All browsers now support ESM, so Astro is able to ship this code directly to the browser during development.

Reviewer 2: ```suggestion
ESM is now native to all browsers, this lets Astro ship code directly to the browser during development, using Astro's own Hot-Module Reloading (HMR).
```

Author: Changed! Better, thanks! (But, it's 2 diff sentences, so I just edited myself.)
